virtool/blast/task.py

```diff
@@ -53,45 +53,69 @@ async def request(self):
 
         self.rid = blast.rid
 
-    async def wait(self):
+    async def wait(self, retries: int = 3):
         """
         Wait until the BLAST search completes.
 
         Keep check the BLAST status on NCBI with increasingly longer intervals between
         checks.
 
-        Checks are conducted by the data layer and will store the results or error when
-        the search completes. The task completes when either an error or
+        Checks are conducted by the data layer and will store the results or error
+        when the search completes. The task completes when either an error or
 
+        The BLAST will be retried up to 3 times if a single BLAST
+        search exceeds 10 minutes
         """
-        try:
-            while True:
-                await asyncio.sleep(self.interval)
 
-                blast = await self.data.blast.check_nuvs_blast(
+        async def wait_with_timeout():
```

Comment:
The logic here feels a bit more side-effect prone than it needs to be. I think in this case it would make more sense to remove `wait_with_timeout` from the class and reduce the scope of what it does.  Have it focus exclusively on the core loop along the lines of:


 1.  Wait for an the starting interval
 2. check if the blast is ready
      - if error or ready, return the result
      - else goto next step
 3. increment the interval
 4. goto step 1

Move all the extra logic of parsing what the returned result means or handling errors to the calling loop.